moveit_core/collision_detection_fcl/test/test_fcl_env.cpp

```diff
@@ -0,0 +1,307 @@
+  double joint4 = -2.356;
+  double joint6 = 1.571;
+  double joint7 = 0.785;
+  panda_state.setJointPositions("panda_joint2", &joint2);
+  panda_state.setJointPositions("panda_joint4", &joint4);
+  panda_state.setJointPositions("panda_joint6", &joint6);
+  panda_state.setJointPositions("panda_joint7", &joint7);
+  panda_state.update();
+}
+
+class CollisionDetectionEnvTest : public testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    robot_model_ = moveit::core::loadTestingRobotModel("panda");
+    robot_model_ok_ = static_cast<bool>(robot_model_);
+
+    acm_.reset(new collision_detection::AllowedCollisionMatrix(robot_model_->getLinkModelNames(), false));
+
+    acm_->setEntry("panda_link0", "panda_link1", true);
+    acm_->setEntry("panda_link1", "panda_link2", true);
+    acm_->setEntry("panda_link2", "panda_link3", true);
+    acm_->setEntry("panda_link3", "panda_link4", true);
+    acm_->setEntry("panda_link4", "panda_link5", true);
+    acm_->setEntry("panda_link5", "panda_link6", true);
+    acm_->setEntry("panda_link6", "panda_link7", true);
+    acm_->setEntry("panda_link7", "panda_hand", true);
+    acm_->setEntry("panda_hand", "panda_rightfinger", true);
+    acm_->setEntry("panda_hand", "panda_leftfinger", true);
+    acm_->setEntry("panda_rightfinger", "panda_leftfinger", true);
+    acm_->setEntry("panda_link5", "panda_link7", true);
+    acm_->setEntry("panda_link6", "panda_hand", true);
+
+    c_env_.reset(new collision_detection::CollisionEnvFCL(robot_model_));
+
+    robot_state_.reset(new robot_state::RobotState(robot_model_));
+
+    setToHome(*robot_state_);
+  }
+
+  void TearDown() override
+  {
+  }
+
+protected:
+  bool robot_model_ok_;
+
+  robot_model::RobotModelPtr robot_model_;
+
+  // std::shared_ptr<collision_detection::CollisionEnvBullet> c_env_;
+  collision_detection::CollisionEnvPtr c_env_;
+
+  collision_detection::AllowedCollisionMatrixPtr acm_;
+
+  robot_state::RobotStatePtr robot_state_;
+};
+
+/** \brief Correct setup testing. */
+TEST_F(CollisionDetectionEnvTest, InitOK)
+{
+  ASSERT_TRUE(robot_model_ok_);
+}
+
+/** \brief Tests the default values specified in the SRDF if they are collision free. */
+TEST_F(CollisionDetectionEnvTest, DefaultNotInCollision)
+{
+  collision_detection::CollisionRequest req;
+  collision_detection::CollisionResult res;
+  c_env_->checkSelfCollision(req, res, *robot_state_, *acm_);
+  ASSERT_FALSE(res.collision);
+}
+
+/** \brief A configuration where the robot should collide with itself. */
+TEST_F(CollisionDetectionEnvTest, LinksInCollision)
+{
+  // Sets the joint values to zero which is a colliding configuration
+  robot_state_->setToDefaultValues();
+  robot_state_->update();
+
+  collision_detection::CollisionRequest req;
+  collision_detection::CollisionResult res;
+  c_env_->checkSelfCollision(req, res, *robot_state_, *acm_);
+  ASSERT_TRUE(res.collision);
+}
+
+// TODO: Add collision check capability within world itself and then enable test
```

Comment:
add your usename to the TODO, or a github issue id

Reply:
I removed this test, see #1565 for reference.